frontend/src/modules/projects/views/projectsView.js

```diff
@@ -1,280 +1,198 @@
-      this.tags.push(filterType);
-      this.filterCollection();
-    },
-
-    removeTag: function(filterType) {
-      // remove filter from this.filters
-      this.tags = _.filter(this.tags, function(item) { return item != filterType; });
-      this.filterCollection();
-    },
-
-    filterCollection: function() {
-      this.search.tags = this.tags.length
-          ? { $all: this.tags }
-          : null ;
-      this.updateCollection(true);
-    },
-
-    convertFilterTextToPattern: function(filterText) {
-      var pattern = '.*' + filterText.toLowerCase() + '.*';
-      return { title: pattern};
+    doLayout: function(layout) {
+      if(this.supportedLayouts.indexOf(layout) === -1) {
+        return;
+      }
+      this.getProjectsContainer().attr('data-layout', layout);
+      this.setUserPreference('layout', layout);
+    },
+
+    doSort: function(sort, fetch) {
+      switch(sort) {
+        case "desc":
+          this.sort = { title: -1 };
+          break;
+        case "updated":
+          this.sort = { updatedAt: -1 };
+          break;
+        case "asc":
+        default:
+          sort = "asc";
+          this.sort = { title: 1 };
+      }
+      this.setUserPreference('sort', sort);
+      if(fetch !== false) this.resetCollection();
     },
 
-    filterBySearchInput: function (filterText) {
-      this.filterText = filterText;
-      this.search = this.convertFilterTextToPattern(filterText);
-      this.setUserPreference('search', filterText);
-      this.updateCollection(true);
-    },
+    doFilter: function(text, tags, fetch) {
+      text = text || '';
+      this.filterText = text;
+      this.search = this.convertFilterTextToPattern(text);
+      this.setUserPreference('search', text);
 
-    filterCoursesByTags: function(tags) {
-      this.setUserPreference('tags', tags);
+      tags = tags || [];
       this.tags = _.pluck(tags, 'id');
-      this.updateCollection(true);
-    },
+      this.setUserPreference('tags', tags);
 
-    setupLazyScrolling: function() {
-      var $projectContainer = $('.projects');
-      var $projectContainerInner = $('.projects-inner');
-      // Remove event before attaching
-      $projectContainer.off('scroll');
+      if(fetch !== false) this.resetCollection();
+    },
 
-      $projectContainer.on('scroll', _.bind(function() {
-        var scrollTop = $projectContainer.scrollTop();
-        var scrollableHeight = $projectContainerInner.height();
-        var containerHeight = $projectContainer.height();
-        // If the scroll position of the assets container is
-        // near the bottom
-        if ((scrollableHeight-containerHeight) - scrollTop < 30) {
-          if (!this.isCollectionFetching) {
-            this.isCollectionFetching = true;
-            this.lazyRenderCollection();
-          }
-        }
-      }, this));
+    onResize: function() {
+      // we don't want to re-initialise for _every_ resize event
+      if(this.resizeTimer) {
```

Comment:
could you use _.throttle for this?

Reply:
Yep, certainly could. Will have a look :)